### PR TITLE
Add support for Play Games provider

### PR DESCRIPTION
**Pull request by SUPERCILEX**
More work on #1196.

For some reason, I just can't get login to work with Play Games. I'm always getting error code 12501 and logs like this: `GamesAuthenticator: Must have a game ID to sign in!`. Normal Google sign-in works just fine and I've tried hardcoding the client id, messing with SHA1 certificates, everything! Could you perhaps try it with your setup to see if it's just me? 🤔

### CODE REVIEW

**Comment by SUPERCILEX** [comment]
@samtstern Do you want to take a look at this one? I'm gonna be honest, I couldn't get it to work, see https://github.com/firebase/FirebaseUI-Android/pull/1211#issue-177305878.

**Comment by samtstern** [comment]
@SUPERCILEX I'll see if I can get games sign in going.

**Comment by samtstern** [comment]
@SUPERCILEX update: I did look at this but I am also having a hell of a time actually getting set up with Play Games ... and I used to work on that product!

File: auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java [hunk 1]
```diff
@@ -119,6 +120,7 @@ private void populateIdpList(List<IdpConfig> providerConfigs,
             @LayoutRes int buttonLayout;
             switch (idpConfig.getProviderId()) {
                 case GoogleAuthProvider.PROVIDER_ID:
+                case PlayGamesAuthProvider.PROVIDER_ID:
```
Comment (samtstern) [inline auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java]: I think we want to have a green "Sign in with Play games" button.  I'll see if I can get us the assets.

**Comment by samtstern** [comment]
Got some assets for you, let me know if we need different ones.

image

[play-games-icons.zip](https://github.com/firebase/FirebaseUI-Android/files/1956518/play-games-icons.zip)

**Comment by SUPERCILEX** [comment]
Sweet, thanks for the assets! And yeah, let me know if you ever find someone who's able to get Play Games sign-in to work 😂

**Comment by samtstern** [comment]
@SUPERCILEX offended if I kick this and Github to `4.1.0`?  We've got a lot in flight right now and I don't think we'll get this working in time, and the TOS/PP changes have a pretty hard deadline with GDPR and whatnot.

**Comment by SUPERCILEX** [comment]
Not at all! That makes sense to me. 👍

**Comment by samtstern** [comment]
I still can't get PGS sign in to work ... so bumping to 4.2.0.  Sorry @SUPERCILEX I know you worked hard on this!

**Comment by SUPERCILEX** [comment]
Haha, well if no one can get this working, maybe you could find someone who worked on it and show us how it's done. 😁

**Comment by pushbit** [comment]
Any luck getting help from an insider?  Years ago I was trying to implement GCM device group messaging and after following the documentation to the letter it just wouldn't work.  Many months later a Googler replied on Stack Overflow (or similar) and revealed that the request needs an undocumented parameter, and then it finally worked.  I get the feeling this is a similar situation.

**Comment by samtstern** [comment]
Still having no luck here.  My attempts fail with an error in the developer console, and a lot of times in those cases being an insider is the _problem_ not the solution.

**Comment by samtstern** [comment]
Sigh.  Booting this from the milestone again.

**Comment by samtstern** [comment]
@SUPERCILEX going to close this old PR so that there's not confusion like there was on #309

**Comment by SUPERCILEX** [comment]
Sounds good!

**Pull request closed without merging**